src/Core/Package.vala

```diff
@@ -597,7 +597,21 @@ public class AppCenterCore.Package : Object {
             return app_info != null;
         }
 
-        string? desktop_id = component.get_desktop_id ();
+        var launchables = component.get_launchable (AppStream.LaunchableKind.DESKTOP_ID).get_entries ();
+        for (int i = 0; i < launchables.length; i++) {
+            app_info = new DesktopAppInfo (launchables[i]);
+            if (app_info != null) {
```

Comment:
Won't `app_info` always be non-null here, causing the break in the very first iteration of this loop?

Reply:
Oops, yes! I need to do a lookup to see if that desktop file exists and break if it does. Thank you for spotting that.

Reply:
Actually, this works exactly as intended.

It's a bit weird but the `DesktopAppInfo` constructor does indeed return `null` if the desktop file isn't found: https://valadoc.org/gio-unix-2.0/GLib.DesktopAppInfo.DesktopAppInfo.html.

I'll leave a comment in the code because it does look strange.